storage/innobase/dict/dict0dict.cc

```diff
@@ -914,6 +914,43 @@ dict_table_autoinc_unlock(
 {
 	mutex_exit(table->autoinc_mutex);
 }
+
+/********************************************************************//**
+Create and initialize the analyze index lock for a given table.
+@param[in,out]	table_void	table whose autoinc latch to create
+*/
+static
+void
+dict_table_analyze_index_alloc(
+	void*	table_void)
+{
+	dict_table_t*	table = static_cast<dict_table_t*>(table_void);
+	table->analyze_index_mutex = UT_NEW_NOKEY(AnalyzeIndexMutex());
+	ut_a(table->analyze_index_mutex != NULL);
+	mutex_create(LATCH_ID_ANALYZE_INDEX_MUTEX, table->analyze_index_mutex);
+}
+
+/********************************************************************//**
+Acquire the analyze index lock. */
+void
+dict_table_analyze_index_lock(
+	dict_table_t*	table)	/*!< in/out: table */
+{
+	os_once::do_or_wait_for_done(
```

Comment:
assert that mutex is not created

Reply:
The assert is already in dict_table_analyze_index_alloc()